alica_engine/include/engine/AlicaContext.h

```diff
@@ -173,6 +173,7 @@ class AlicaContext
      * @note Can throw an exception if some necessary configurations dont exist in the config file.
      */
     AlicaContext(const AlicaContextParams& alicaContextParams);
+    AlicaContext(const AlicaContextParams&& alicaContextParams);
```

Comment:
I don't know or even noticed that "syntax rule" about bound const& to lvalue without using the move constructor.
However, I have done this modification because the test won't link. I checked again this evening.  
These work
`AlicaContext(const AlicaContextParams alicaContextParams);` 
`AlicaContext(const AlicaContextParams&& alicaContextParams);` 
Perhaps I have just done an error aligning the method in .cpp.